internal/core/application/market_loader_service_test.go

```diff
@@ -0,0 +1,225 @@
+package application
+
+import (
+	"context"
+	"github.com/stretchr/testify/require"
+	"github.com/tdex-network/tdex-analytics/internal/core/domain"
+	"github.com/tdex-network/tdex-analytics/internal/infrastructure/db/inmemory"
+	"testing"
+)
+
+var (
+	ctx     context.Context
+	filter1 = domain.Filter{
+		Url:        "url1",
+		BaseAsset:  "ba1",
+		QuoteAsset: "qa1",
+	}
+	filter2 = domain.Filter{
+		Url:        "url2",
+		BaseAsset:  "ba2",
+		QuoteAsset: "qa2",
+	}
+	filter3 = domain.Filter{
+		Url:        "url3",
+		BaseAsset:  "ba3",
+		QuoteAsset: "qa3",
+	}
+	page = domain.Page{
+		Number: 1,
+		Size:   20,
+	}
+)
+
+func TestUpdateMarketActiveStatusAndInsertNew(t *testing.T) {
+	type fields struct {
+		marketRepository domain.MarketRepository
+	}
+	type args struct {
+		activeMarkets []domain.Market
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		wantErr     bool
+		prepareData func(repo domain.MarketRepository) ([]domain.Market, error)
+		validate    func(t *testing.T, repo domain.MarketRepository)
+	}{
+		{
+			name: "test_1",
```

Comment:
we should give these tests more descriptive name to give clue about what case they are testing.
The name for this test, for example, could be:
```suggestion
			name: "add_market_if_not_existing",
```

Reply:
maybe you could name this one but when you have more test cases you are facing a problem of finding good name.
it is necessary to see passed args(existing, active) and validate method to find out what was the goal.

Reply:
If we add other test cases, they will all have a specific name.

I don't see finding a good name as a real problem because when someone writes a tests, he does really well know which case he's testing.

Reply:
> it is necessary to see passed args(existing, active) and validate method to find out what was the goal.

the goal of the name is exactly to prevent somebody to be forced to look at the code to understand which cases have been tested.

Reply:
Done